### ISSUE TITLE
PEP8-aware mode for `af`/`ac`

### ISSUE THREAD

**Issue by lieryan** [bps/vim-textobj-python#26]
I'm proposing that `daf`/`dac` should be written in such a way that it should leave the amount of black lines after cutting the text to be PEP8 compliant. 

PEP8 requires that top level classes/functions have two blank lines while leaving a single blank line in other cases.

Having the text object be PEP8-compliant would make it a lot easier to move things around as sequences like `daf` `]pf` `P` would just work as expected without needing to fiddle with fixing blank lines. This also matches the behaviour of built-in text objects like `daw` `w` `P`. 

Open questions:

1. Should this be default behaviour or an option?
1. If I create a PR for this, does it have a chance to be accepted?

**Comment by bps** [comment]
Sounds reasonable to me. Re: (1), the default behavior should match what the built in C-style objects do. If this behavior more closely matches those, then it should be the default. Re: (2), as long as you write test cases, I’ll merge it.

**Comment by lieryan** [comment]
@bps thanks for your reply, I've created PR https://github.com/bps/vim-textobj-python/pull/27, which probably actually more closely resembles the behaviour proposed in https://github.com/bps/vim-textobj-python/issues/17 rather than the one in this ticket.

If I ever get around to implementing strict PEP8 selection behaviour I proposed here, I would likely keep it behind an option, because after thinking through the various cases, I think strict PEP8 whitespace handling behaviour would necessarily have many corner cases that would be surprising for experienced vim users, particularly around the behaviour when the original code spacing wasn't very PEP8 compliant to begin with.

For now, I'm ok with the behaviour on https://github.com/bps/vim-textobj-python/pull/27, so I'm happy to close this for now.
